gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -1204,83 +1183,39 @@ public int compare(final User u1, final User u2) {
-	 * So for a given user editing many cells there will be many GbGradeCells associated with that user. These have a time associated with
-	 * each so we can discard manually if desired, on lookup.
-	 *
-	 * @param gradebookUid
-	 */
-	private void pushEditingNotification(final String gradebookUid, final User currentUser, final String studentUuid,
-			final long assignmentId) {
-
-		// TODO Tie into the event system so other edits also participate in
-		// this
-
-		// get the notifications for this gradebook
-		Map<String, Map<String, GbGradeCell>> notifications = (Map<String, Map<String, GbGradeCell>>) this.cache
-				.get(gradebookUid);
-
-		Map<String, GbGradeCell> cells = null;
-
-		// get or create cell map
-		if (notifications != null) {
-			cells = notifications.get(currentUser.getId());
-		} else {
-			notifications = new HashMap<>();
-		}
-
-		if (cells == null) {
-			cells = new LinkedHashMap<>();
-		}
-
-		// push the edited cell into the map. It will add/update as required
-		cells.put(buildCellKey(studentUuid, assignmentId),
-				new GbGradeCell(studentUuid, assignmentId, currentUser.getDisplayName()));
-
-		// push the new/updated cell map into the main map
-		notifications.put(currentUser.getEid(), cells);
-
-		// update the map in the cache
-		this.cache.put(gradebookUid, notifications);
-
-	}
-
-	/**
-	 * Get a list of editing notifications for this gradebook. Excludes any notifications for the current user
+	 * Get a list of edit events for this gradebook. Excludes any events for the current user
 	 *
 	 * @param gradebookUid the gradebook that we are interested in
+	 * @param since the time to check for changes from
 	 * @return
 	 */
-	public List<GbGradeCell> getEditingNotifications(final String gradebookUid) {
-
-		final String currentUserId = getCurrentUser().getEid();
+	public List<GbGradeCell> getEditingNotifications(final String gradebookUid, final Date since) {
 
-		// get the notifications for this gradebook
-		Map<String, Map<String, GbGradeCell>> notifications = (Map<String, Map<String, GbGradeCell>>) this.cache
-				.get(gradebookUid);
+		final User currentUser = getCurrentUser();
 
 		final List<GbGradeCell> rval = new ArrayList<>();
 
-		if (notifications != null) {
-			// clone array
-			notifications = new HashMap(notifications);
-			notifications.remove(currentUserId);
-
-			// join the rest of the maps to get a flat list of GbGradeCells
-			for (final Map<String, GbGradeCell> cells : notifications.values()) {
-				rval.addAll(cells.values());
+		final List<Assignment> assignments = getGradebookAssignments(gradebookUid);
+		final List<Long> assignmentIds = assignments.stream().map(a -> a.getId()).collect(Collectors.toList());
+		final List<GradingEvent> events = this.gradebookService.getGradingEvents(assignmentIds, since);
+
+		// filter out any events made by the current user
+		final Map<String, GbUser> userCache = new HashMap<>();
```

Comment:
Yeah, this was just to ensure we only need to hit `getUser` once for each distinct user as we iterate over the events - this "cache" is per-request/method call so doesn't need to be cluster safe... perhaps the `userCache` attribute name is a little misleading!